File: OOP_HW6/main.py
# This is a sample Python script.

# Press Shift+F10 to execute it or replace it with your code.
# Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.
from abc import ABCMeta, abstractmethod


class Shape:
    def __init__(self, color):
        self.color = color

    def print(self):
        print(self.color, end=' ')

    @abstractmethod
    def get_area(self):
        pass


class Circle(Shape):
    def __init__(self, color, radius):
        super().__init__(color)
        self.radius = radius

    def print(self):
        super().print()
        print(" Circle, radius" + str(self.radius) + " ,area " + str(self.get_area()))

    def get_area(self):
        return self.radius * self.radius * 3.14


class Rectangle(Shape):
    def __init__(self, color, height, width):
        super().__init__(color)
        self.height = height
        self.width = width

    def print(self):
        super().print()
        print(" Rectangle, height " + str(self.height) + " ,area " + str(self.get_area()))

    def get_area(self):
        return self.height * self.width


class Triangle(Shape):
    def __init__(self, color, height, base):
        super().__init__(color)
        self.height = height
        self.base = base

    def print(self):
        super().print()
        print(" Triangle, height " + str(self.height) + ", base " + str(self.base) + ", area " + str(self.get_area()))

    def get_area(self):
        return 0.5 * self.height * self.base


def main():
    shape = []
    classtype = []
    x, y = 10, 20
    s = "yellow"
    circle = Circle(s, x)
    a = Triangle(s, x, y)
    b = Rectangle(s, x, y)
    c = Triangle("red", 3, 5)
    d = Circle("green", 5)
    e = Rectangle("white", 3, 8)
    shape.append(circle)
    classtype.append("circle")
    shape.append(a)
    classtype.append("triangle")
    shape.append(b)
    classtype.append("rectangle")
    shape.append(c)
    classtype.append("triangle")
    shape.append(d)
    classtype.append("circle")
    shape.append(e)
    classtype.append("rectangle")
    print("Printing all shapes:")
    for i in shape:
        i.print()

    print("Print all circle:")
    for i in shape:
        if type(i) == type(circle):
            i.print()

    for i in shape:
        del i

    return 0


main()

File: OOP_HW6/main.java
import java.util.*;
public class main{
    public static void main(String[] args){
        Vector<Shape> s = new Vector<>();
        s.add(new Triangle("green",8,4));
        s.add(new Triangle("Red",4,10));
        s.add(new Circle("yellow",5));
        s.add(new Circle("blue",10));
        s.add(new Rectangle("black",8,6));
        s.add(new Rectangle("pink",3,7));
        s.add(new Circle("purple",43));
        s.add(new Triangle("indigo",53,97));
        s.add(new Rectangle("orange",103,886));
        s.add(new Circle("white",87));
        s.add(new Circle("brown",69));
        System.out.println("Printing all shapes...\n");
        for(int i=0;i<s.size();i++){
            s.get(i).print();
        }
        System.out.println("Printing only circles...\n");
        Circle circle = new Circle("circle",5);
        for(int i=0;i<s.size();i++){
            if(s.get(i).getClass().getName().equals(circle.getClass().getName())){
                s.get(i).print();
            }
        }
        System.out.println("Destructing every Objects...\n");
        s.clear();

    }
}
